app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddSidewalkSurfaceForm.kt

```diff
@@ -76,167 +43,79 @@ class AddSidewalkSurfaceForm : AbstractQuestFormAnswerFragment<SidewalkSurfaceAn
-                        }.show()
-                    }
-                    .setNegativeButton(android.R.string.cancel, null)
-                    .show()
-            } else {
-                onSelectedSide(SurfaceAnswer(it.value!!), isRight)
-            }
-        }.show()
+    override fun serializeAnswer(answer: LastSelection<SurfaceAnswer>): String {
+        return "${answer.left.value.value}#${answer.left.value.note}#${answer.right.value.value}#${answer.right.value.note}"
     }
 
-    private fun onSelectedSide(surface: SurfaceAnswer, isRight: Boolean) {
-        val image = DrawableImage(RotatedCircleDrawable(resources.getDrawable(surface.value.asItem().drawableId!!)))
+    override fun deserializeAnswer(str: String): LastSelection<SurfaceAnswer> {
+        val arr = str.split('#')
 
-        if (isRight) {
-            binding.puzzleView.replaceRightSideFloatingIcon(image)
-            binding.puzzleView.replaceRightSideImage(ResImage(R.drawable.ic_sidewalk_illustration_yes))
-            rightSide = surface
+        val leftSurface = Surface.values().find { it.toString() == arr[0] }!!
+        val leftNote = if (arr[1] == "null") {
+            null
         } else {
-            binding.puzzleView.replaceLeftSideFloatingIcon(image)
-            binding.puzzleView.replaceLeftSideImage(ResImage(R.drawable.ic_sidewalk_illustration_yes))
-            leftSide = surface
+            arr[1]
         }
-        updateLastAnswerButtonVisibility()
-        checkIsFormComplete()
-    }
-
-    /* --------------------------------- last answer button ------------------------------------- */
-
-    private fun initLastAnswerButton() {
-        updateLastAnswerButtonVisibility()
-
-        lastSelection?.let {
-            binding.lastAnswerButton.leftSideImageView.setImageResource(it.left.value.asItem().drawableId!!)
-            binding.lastAnswerButton.rightSideImageView.setImageResource(it.right.value.asItem().drawableId!!)
-        }
-
-        binding.lastAnswerButton.root.setOnClickListener { applyLastSelection() }
-    }
-
-    private fun updateLastAnswerButtonVisibility() {
-        val formIsPrefilled = leftSide != null || rightSide != null
-        val lastAnswerWasForBothSides = (lastSelection?.left != null && lastSelection?.right != null)
-        val isDefiningBothSides = isDefiningBothSides && lastAnswerWasForBothSides
-
-        binding.lastAnswerButton.root.isGone =
-            lastSelection == null || formIsPrefilled || !isDefiningBothSides
-    }
-
-    private fun saveLastSelection() {
-        val leftSide = leftSide
-        val rightSide = rightSide
-        if (leftSide != null && rightSide != null) {
-            lastSelection =
-                if (isRoadDisplayedUpsideDown())
-                    LastSidewalkSurfaceSelection(rightSide, leftSide)
-                else
-                    LastSidewalkSurfaceSelection(leftSide, rightSide)
-        }
-    }
+        val leftStreetSideItem = StreetSideItem2(
```

Comment:
This feels like quite a bit of duplication here, but the only other way I could think of was to change `note` in `SurfaceAnswer` to `var`, which I wasn't sure was a good idea. (Then it could be done same as in sidewalk and shoulder with `items.find { it.value.toString() == arr[0] }!` and then just set the note value and return that.)

Reply:
(not a very good quote there, sorry, I mean line 97 to 103, and same for right)